src/pages/index.js

```diff
@@ -1,106 +1,226 @@
 
-const userInfo = new UserInfo({
-  nameSelector: '.profile__title', 
-  jobSelector: '.profile__description'
-});
-
-const editProfilePopup = new PopupWithForm(profileEditModal, ({ title, description }) => {
-  userInfo.setUserInfo({ title, job: description });
-  editProfilePopup.close();
+const api = new Api({
+  baseURL: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "d8b9199f-b9d7-4b7f-ad09-c5597d55941e",
+    "Content-Type": "application/json",
+  },
 });
 
-profileEditBtn.addEventListener('click', openProfilePopup);
-
-function openProfilePopup() {
-  const { name, job } = userInfo.getUserInfo();
-  profileTitleInput.value = name;
-  profileDescriptionInput.value = job;
-  editFormValidator.toggleButtonState();
-  editProfilePopup.open();
-}
-
-editProfilePopup.setEventListeners();
+let sectionEl;
+let currentUserId;
+
+api
+  .getAppInfo()
+  .then(([userData, cardData]) => {
+    currentUserId = userData._id;
+    userInfoEl.setUserInfo({ name: userData.name, job: userData.about });
+    userInfoEl.setAvatarInfo(userData.avatar);
+
+    sectionEl = new Section(
+      {
+        items: cardData,
+
+        renderer: (data) => {
+          const card = createCard(data);
+          sectionEl.addItem(card);
+        },
+      },
+      ".cards__list"
+    );
+    sectionEl.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
 
 /* -------------------------------------------------------------------------- */
-/*                                Section Class                               */
+/*                                    Card                                    */
 /* -------------------------------------------------------------------------- */
 
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: ({ name, link }) => {
-      const newCard = createCard({ name, link });
-      cardListSection.addItem(newCard);
+function createCard(data) {
+  const card = new Card({
+    data,
+    cardSelector: "#card-template",
+    currentUserId,
+    handlePreviewImage: () => {
+      imagePopup.open(data);
     },
-  },
-  cardListEl
-);
-
-cardListSection.renderItems();
+    handleLikeCard: (shouldRemoveLike) => {
+      if (shouldRemoveLike) {
+        api
+          .removeLike(data)
+          .then((data) => {
+            card.updateLikes(data.likes);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      } else {
+        api
+          .addLike(data)
+          .then((data) => {
+            card.updateLikes(data.likes);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      }
+    },
+    handleDeleteCard: (cardId) => {
+      deletePopup.open();
+      deletePopup.setSubmitAction(() => {
+        deletePopup.renderLoading(true);
+        api
+          .deleteCard(cardId)
+          .then(() => {
+            card.deleteClick();
```

Comment:
Please rename `deleteClick` to something like `deleteCard`, since it's not a click event listener anymore